Some docs cleanup

bollwyvl opened the pull request:
## References

- bumps the frontend architecture to reflect the changes from #165
- some things missing from #177, like how to build the docs

## Code changes

- travis
  - `3.8` (not `-dev`)
  - ~~add `3.9-dev`~~
    - > still getting the odd `ModuleNotFoundError: No module named 'pip'`

## User-facing changes

Some more progress towards getting the docs usable, fixing links, etc.

## Backwards-incompatible changes

N/A

## Chores

- [x] linted
- [ ] tested
- [x] documented
- [ ] changelog entry

bollwyvl commented:
Actually got the link checker wired up and passing locally... once this works in CI, that's probably a good place to stop!

bollwyvl commented:
LGTM

krassowski commented on docs/EXTENDING.md:
```diff
@@ -13,7 +13,7 @@
 or distributed by third parties as python or JSON files. Since we'd like to see
 as many Language Servers work out of the box as possible, consider
-[contributing a spec](../CONTRIBUTING.md#specs), if it works well for you!
+[contributing a spec](../CONTRIBUTING.ipynb#specs), if it works well for you!
```
Confused by this...

bollwyvl replied:
fixed

krassowski commented on docs/LANGUAGESERVERS.md:
```diff
@@ -42,12 +42,12 @@ tested to work with `jupyter-lsp`.
 [bring your own language server](#adding-custom-language-servers).
 
-> Please consider [contributing your language server spec](../../CONTRIBUTING.md#spec)
+> Please consider [contributing your language server spec](../Contributing.md)
```
this

bollwyvl replied:
fixed

krassowski commented on docs/LANGUAGESERVERS.md:
```diff
@@ -104,32 +106,32 @@ c.LanguageServerManager.language_servers = {
 `pip`-installable packages in the same environment as the Jupyter `notebook` server
 can be automatically detected as providing a language server spec. These are a
-little more involved: see [CONTRIBUTING](../../CONTRIBUTING.md).
+little more involved: see [CONTRIBUTING](Contributing.md).
```
and this.

bollwyvl replied:
fixed

krassowski commented on docs/conf.py:
```diff
@@ -88,10 +88,10 @@
 
 html_context = {
-    "display_github": True,
+    "display_github": False,
```
IMO worth turning on (not necessarily in this PR)

bollwyvl replied:
turned on

krassowski commented:
Looks good in general, just want to get a confirmation about the paths to Contributing. By my intuition, at least one of them is must be wrong (but I can be wrong too!).

bollwyvl commented:
All of the above are link-checker-driven-development artifacts, whoops:
- i think i can fix all the internal links: (nb)sphinx is strangely passive aggressive about file extensions, in that sometimes it doesn't even care which you use, and sometimes gets very angry
  - might end up rewriting more markdown getting passed through to notebooks, or see if i can get rid of the wrappers altogether (preferable)
- the github link creates links on generated pages that don't exist, like `search` and `genindex`
  - i'll look into not showing those on those specific pages

bollwyvl commented:
Well, that ended up being a bit more work than i was hoping, but it's feeling better:
- fixed the the `search` and `genindex` github links 
  - ...rather brutally by forking the template (ignored by `prettier`)
- moved all the `docs` content into notebooks
- generate the language server tables directly from a hot manager instance :tada: 
  - requires doing the full yarn and python install, but we'll KNOW the docs are right (if they build)

I guess at this point with as much content in there as there is, it may well be worth it to grab some of the stuff from `prenotebook` as part of `lint.py`:
- `black`/`isort` the python code cells
- `prettier` the markdown
- strip the outputs/execution numbers

bollwyvl commented on docs/_templates/breadcrumbs.html:
```diff
@@ -0,0 +1,82 @@
+    {% endblock %}
+    {% block breadcrumbs_aside %}
+      <li class="wy-breadcrumbs-aside">
+        {% if hasdoc(pagename) and pagename not in ["search", "genindex"] %}
```
all for this line...

krassowski commented on README.md:
```diff
@@ -4,11 +4,11 @@
 > _This project is still maturing, but you are welcome to check it out, leave feedback and/or a PR_
 
-Quick Links: **[Installation](#installation) | [Language Servers](./docs/LANGUAGESERVERS.md) | [Updating](#updating) | [Changelog](./CHANGELOG.md) | [Roadmap](./docs/ROADMAP.md) | [Contributing](./CONTRIBUTING.md) | [Extending](./docs/EXTENDING.md)**
+Quick Links: **[Installation](./Installation.ipynb) | [Configuring](./docs/Configuring.ipynb) | [Updating](#updating) | [Changelog](./CHANGELOG.md) | [Roadmap](./docs/Roadmap.ipynb) | [Contributing](./CONTRIBUTING.md) | [Extending](./docs/Extending.ipynb)**
```
`./docs/Installation.ipynb` ? Actually, I would probably prefer to have this go to the short version of the installation within README (we can shorten it) and then to have a link to the notebook from here.

In the quick links we maybe could have something like: `Installation: [Basic](#installation) | [Full](notebook)` (but I am not full convinced)

bollwyvl replied:
agreed, README.md needs to pretty much be standalone for the 80% case. Hopefully fixed now.

bollwyvl commented:
Thanks for the review, looking into it now...

bollwyvl commented:
Probably best if this lands prior to #199 so that i can fix up changelog and extending while it's still fresh in my mind... so many moving pieces.

krassowski commented:
We will need to do something about these Linux/Docker failures...

krassowski commented on README.md:
````diff
@@ -78,26 +78,60 @@ Either:
+   ```bash
+   conda install -c conda-forge nodejs
+   # or one of the following, as an administrator
+   choco install nodejs            # Windows with Chocolatey
````
if it is all in one block, I am 100% sure some users will just copy-paste it all. I would move these examples to the docs.

krassowski commented on README.md:
````diff
@@ -78,26 +78,60 @@ Either:
+   ```bash
+   conda install -c conda-forge python=3
+   # or
+   conda install -c conda-forge jupyterlab=1.2
````
?
